---
title: "geneAnnotations"
author: "Katriona Goldmann"
date: "3/7/2021"
output: md_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(kableExtra)
library(omicAnnotations)
```


# omicAnnotations

This package pools together information from different databases and APIs in order to annotate SNPs and genes. In particular this uses databases by:

<a href="https://www.ebi.ac.uk/"><img src="./images/ebi.png" alt="EBI" height="70" style="margin:5px;"/></a>&nbsp;<a href="https://www.qmul.ac.uk/whri/emr/"><img src="./images/emr.png" alt="EMR" height="70" style="margin:5px;"/></a>&nbsp;<a href="https://maayanlab.cloud/Enrichr/"><img src="./images/enrichr.png" alt="enrichr" height="70" style="margin:5px;"/></a>&nbsp;<a href="https://www.ncbi.nlm.nih.gov/"><img src="./images/ncbi.png" alt="ncbi" height="70" style="margin:5px; margin-bottom:15px;"/></a>&nbsp;<a href="https://www.gtexportal.org/home/"><img src="./images/gtex.png" alt="gtex" height="50" style="margin:5px;"/></a> &nbsp;<a href="https://pubmed.ncbi.nlm.nih.gov/"><img src="./images/pubmed.png" alt="pubmed" height="70" style="margin:5px;"/></a>&nbsp;



To install: 

```{r, eval=FALSE}
devtools::install_github("KatrionaGoldmann/omicAnnotations")
```

## Gene Annotations

For example for the entire gene summary: 

```{r}
gene_df <- gene_summary(genes=c("FMOD", "FGF1", "SLAMF6"), diseases="C20")
```


```{r}
kable(gene_df, format = "markdown", row.names = FALSE)
```

### Publications

You can check for publications focusing on genes with given terms. Either 
using `associated_publications`:

```{r}
gene_pubs <- associated_publications(genes=c("FGF1"), 
                                     keywords=c("rheumatoid"), 
                                     split="OR", 
                                     verbose=TRUE)

kable(gene_pubs, format = "markdown", row.names=FALSE)
```


Or `gene_summary`: 

```{r}
gene_df <- gene_summary(genes=c("FGF1"), 
                        associated_diseases = FALSE,
                        gene_description=FALSE, 
                        publications = TRUE)

kable(gene_df, format = "markdown", row.names=FALSE)
```

### Enriched pathways

Looks for enriched pathways with gene sets using [enrichR](https://maayanlab.cloud/Enrichr/).

```{r, eval=FALSE}
lymphoid_pathways <- enriched_pathways(
  genes=c("LAMP5", "LINC01480", "FAM92B", "SLAMF6", "CEP128",
          "FKBP11", "CRTAM", "ISG20", "ZBP1", "TMEM229B",
          "FAM46C", "XBP1", "APOBEC3G", "TNIK", "CD2", "SP140",
          "ACOXL", "PTPRCAP", "PDCD1", "KCNN3", "GZMK",
          "IGFLR1", "SH2D2A", "PIM2", "TPST2"),
  libraries = c('Pathways'),
  dbs=NULL,
  check_for_updates = FALSE)
```

If that doesn't work it may be because the website is down. This happens occasionsally. You can check by using: 

```{r, eval=FALSE}
listEnrichrDbs()
```

```{r, include=FALSE}
data("lymphoid_pathways")
```

Plots

```{r}
lymphoid_pathways$plot
```

### eQTL Catalogue

```{r}
eqtl_table <- associated_eqtl(genes=c("ENSG00000164308"), p_cutoff=1)

kable(eqtl_table, row.names=F) 
```

## SNP Annotations

omicAnnotations can also be used to find out more info about SNPs.

### GWAS Catalogue

```{r}
gwas_traits <- associated_traits(snps = c("rs2910686", "rs7329174"))

kable(gwas_traits, row.names = F) 
```

### eQTL Catalogue

```{r}
eqtl_table <- associated_eqtl(snps = c("rs2910686", "rs7329174"),
                              p_cutoff = 0.05)

kable(eqtl_table, row.names = F) 
```


### GTex 

```{r}
g2s <- data.frame("Genes"=c("ERAP2", "ERAP2", "HLA-DRB9"), 
                  "Snps"=c("chr5_96916728_G_A", "chr5_96916885_T_C", 
                           "chr6_32620055_A_G"))

df <- gtex_eqtl(gene_snp_pairs = g2s)
```

```{r, fig.height=10, message=FALSE}
library(ComplexHeatmap)

hm <- gtex_heatmap(df)
draw(hm, heatmap_legend_side = "left")
```